### ISSUE TITLE
Spark/Keras: checkpoint is only relying on local val loss on GPU 0

### ISSUE THREAD

**Issue by chongxiaoc** [horovod/horovod#3156]
**Bug report:**
When multi-GPUs training is enabled:

- Only GPU 0 is doing checkpoint: https://github.com/horovod/horovod/blob/master/horovod/spark/keras/remote.py#L158
- GPU 0 can only access local validation data:https://github.com/horovod/horovod/blob/master/horovod/spark/keras/remote.py#L231
- Checkpoint is biased by GPU 0's local val data and will be used to overwrite model weights in the end of training: https://github.com/horovod/horovod/blob/master/horovod/spark/keras/remote.py#L263
 

An allreduce scheme should be added to fix local validation issue.

**Comment by chongxiaoc** [comment]
not an issue.
`MetricAverageCallback` is added by default: https://github.com/horovod/horovod/blob/master/horovod/spark/keras/remote.py#L152
